rmw/include/rmw/get_topic_endpoint_info.h

```diff
@@ -31,6 +31,15 @@ extern "C"
  * the associated topic type, the publisher's gid, and the publisher QoS profile.
  * Names of non-existent topics are allowed, in which case an empty array will be returned.
  *
+ * \par QoS that are correctly read
+ *   The QoS profiles returned might have some invalid fields.
+ *   Particularly, the only QoS policies that are guaranteed to be shared
+ *   during discovery are the ones that participate in endpoint matching.
+ *   From the current QoS settings available, the only ones not shared by DDS based
+ *   implementations are `history` and `history_depth`.
+ *   The rmw implementation must set the settings that were not correctly read to
+ *   `RMW_QOS_POLICY_*_UNKNOWN`.
```

Comment:
Here's a suggestion based on my previous review comments:

```suggestion
 * \section QoS that are correctly read
 *
 * The QoS profiles returned may have some invalid fields when the
 * rmw implementation does not have the required information.
 * The rmw implementation must set these settings to
 * `RMW_QOS_POLICY_*_UNKNOWN` in order to indicate this.
 *
 * \subsection Expected behavior for DDS based rmw implementations
 *
 * In particular for DDS based rmw implementations, the only QoS policies
 * that are guaranteed to be shared during discovery are the ones that
 * participate in SEDP endpoint matching.
 * From the current QoS settings available, the only ones not shared by DDS based
 * rmw implementations are `history` and `history_depth`.
```

Reply:
See https://github.com/ros2/rmw/pull/308#discussion_r611952159.